### PR TITLE
enable exporting of users from settings view

### LINKED ISSUE #4260: Export user list action
As an admin of a Tutanota account i want to be able to export the list of my existing users to analyze the data with third party apps.

**Acceptance criteria**
- User list can be exported as csv file
- User data should include name, email address, creation date, deactivation date, used storage, email aliases.

**Implementation hint**
We already have an import action for users which is located next to "Settings > Users management" when the user management is selected. I suggest to include export action there so we have import / export at the same place behind a dropdown.

**Tasks**
- [x] Add dropdown button 
- [x] Load user data 
- [x] Write csv file

### PR DESCRIPTION
**Pull request by johnbotris**
the folder button on "User management" is now a dropdown with 2 children:
import users, export users

export users will automatically render a csv and write it to a file with the format:
name; mail address; created date; deleted date; storage used; aliases

close #4260

### CODE REVIEW

File: src/file/FileController.ts [hunk 1]
```diff
@@ -320,6 +320,11 @@ export class FileController {
 		}
 	}
 
+	async saveStringAsFile(contents: string, name: string, mimeType: string) {
```
Comment (charlag) [inline src/file/FileController.ts:323]: I don't know if it's justified to have this function, we operate on files elsewhere in this class. I'm not insisting tho.

File: src/settings/UserDataExporter.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	const {user} = logins.getUserController()
+	const {userGroups} = await entityClient.load(CustomerTypeRef, assertNotNull(user.customer))
+
+	const localAdminGroupIds = logins
+		.getUserController()
+		.getLocalAdminGroupMemberships()
+		.map(gm => gm.group)
+
+	return promiseMap(
+		(await entityClient.loadAll(GroupInfoTypeRef, userGroups))
```
Comment (charlag) [inline src/settings/UserDataExporter.ts:51]: Maybe introduce a temp variable, it's hard to figure what it is without name. `administeredGroups`?

File: src/settings/UserDataExporter.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+ */
+export async function loadUserExportData(
+	entityClient: EntityClient,
+	userManagementFacade: UserManagementFacade,
+	logins: LoginController,
+): Promise<Array<UserExportData>> {
+	const {user} = logins.getUserController()
+	const {userGroups} = await entityClient.load(CustomerTypeRef, assertNotNull(user.customer))
+
+	const localAdminGroupIds = logins
```
Comment (charlag) [inline src/settings/UserDataExporter.ts:45]: could make a set but eh

File: src/settings/UserDataExporter.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+				// if we are a global admin we keep all group infos
+				// otherwise we only keep group infos of users who the logged in user administrates
+				info => logins.getUserController().isGlobalAdmin()
+					|| (info.localAdmin && localAdminGroupIds.includes(info.localAdmin))
+			),
+		async info => {
+			let usedStorage: number | null = null
+
+			const group = await entityClient.load(GroupTypeRef, info.group)
+			if (group.user != null) {
```
Comment (charlag) [inline src/settings/UserDataExporter.ts:62]: If it's a `userGroup` it should have a user, right? If you want to keep the check maybe it's best to add a comment on why we want to check it

File: src/settings/UserDataExporter.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+export function renderExportDataCsv(userData: Array<UserExportData>) {
+
+	const lines = [
+		USER_EXPORT_CSV_HEADER,
+		...userData.map(
+			data => {
+				const created = formatDateTimeUTC(data.created)
+				const deleted = mapNullable(data.deleted, formatDateTimeUTC)
+				const usedStorage = data.usedStorage ? `${data.usedStorage}MB` : null
```
Comment (charlag) [inline src/settings/UserDataExporter.ts:87]: I don't think it's megabytes

File: src/misc/TranslationKey.ts [hunk 6]
```diff
@@ -1450,4 +1450,6 @@ export type TranslationKeyType =
 	| "yourFolders_action"
 	| "yourMessage_label"
 	| "you_label"
-	| "emptyString_msg"
\ No newline at end of file
+	| "emptyString_msg"
+	| "exportUsers_action"
+	| "importExport_action"
```
Comment (charlag) [inline src/misc/TranslationKey.ts:1455]: I wouldn't add this key, we might add more things into that menu, maybe best just `more_label`

**Review by charlag — CHANGES_REQUESTED** [review]

File: src/file/FileController.ts (hunk 1 above)
Comment (johnbotris) [reply src/file/FileController.ts:323]: i like convenience methods

File: src/settings/UserDataExporter.ts (hunk 5 above)
Comment (johnbotris) [reply src/settings/UserDataExporter.ts:87]: oops true

**Comment by charlag** [comment]
You have `importExport_action` in `de` and probably `de_sie`

**Review by charlag — APPROVED** [review]

**Review by ganthern — APPROVED** [review]

**Pull request merged**